rfcs/0000-sudt-info/0000-sudt info.md

```diff
@@ -0,0 +1,184 @@
+
+For this type of sudt, you can construct an info_cell while minting sudt, whose typescript's code_hash is info_cell typescript, its args is sudt's type_script hash. Info_cell will check whether the transaction meets the conditions.
+
+````
+// Issue new Sudt/Sudt_Info
+Inputs:
+  <... one of the input cells must have owner lock script as lock>
+Outputs:
+	Sudt_Cell:
+		Data:
+			amount: uint128
+        Type:
+        	code_hash: simple_udt type script
+     		args: owner lock script hash (...)
+     	Lock:
+     		<user defined>
+	Sudt_Info_Cell:
+  		Data:
+  			sudt info
+  		Type:
+  			code_hash: sudt_info type script
+  			args: sudt type script hash
+  		Lock:
+  			sudt creator defined
+````
+
+The following rules should be met in a Sudt Info Cell (typescript):
+
+- **Rule 1:** In this transaction, at least one sudt cell must exist in the output cell, and the hash of its typescript matches the args of Info_cell.
+- **Rule 2:** In this transaction, at least one cell must exist in the input cell, and its lockscript is the owner lockscript of sudt.
+- **Rule 3:** the lockscript of info_cell should be set to deadlock by default, in some cases, developers can choose other lockscript.
+- **Rule 4:** If there are multiple Info_cells, choose the info_cell with the smallest block height at the time of generation. If it is at the same block height, choose the info_cell with the smallest transaction index. The same transaction cannot generate two info_cells.
+
+### 3.2. For Script-drive sudt
+
+However, there is another type of sudt on CKB, which is not issued by a lockscript controlled by a private key but follows a specific script logic. Such as NexisDAO's dCKB and Tai, and the previous Info_cell design is no longer suitable for this situation.
+
+Since anyone can mint sudt while following the script logic, if we continue to use the previous logic, anyone can generate the corresponding info_cell. So we need to add new logic on the basis of the previous design.
+
+Since it is impossible to distinguish whether an owner lockscript is script-driven from the outside, it should be checked first according to the script-driven logic.
+
+````
+// Issue new Sudt/Sudt_Info
+Inputs:
+	 ...
+Outputs:
+	Sudt_Info_Cell:
+		Data:
+			sudt info
+		Type:
+			code_hash: sudt_info type script
+			args: type_id
+		Lock:
+			sudt creator defined
+````
+
+The following rules should be met in a Sudt Info Cell (typescript):
+
+- **Rule1**: The cell data conforms to the format.
+- **Rule2**: The args conform to the rules of type id.
+- **Rule 3:** The lockscript of info_cell should be set to deadlock by default, in some cases, developers can choose other lockscript.
+- **Rule 4:** After constructing this Info cell, use the hash of info_cell's type_script as the first 32 bytes of the args of the lockscript of sudt owner. 
```

Comment:
This rule is not safe. When more than two sud owners use the same 32-bytes-prefix, these different sudts will share the same udt info.

Reply:
> This rule is not safe. When more than two sudt owners use the same 32-bytes-prefix, these different sudts will share the same udt info. No rules here can prevent this happen.

I think it's okay for two tokens to have the same name, decimals.

Reply:
Yes, you are right, I have got that.